### PR TITLE
RBE refactoring and updates to support current webrtc compute delta implementation

### PR DESCRIPTION
**Pull request by jcchuks**

### CODE REVIEW

File: src/org/jitsi/impl/neomedia/rtp/remotebitrateestimator/RemoteBitrateEstimatorSingleStream.java [hunk 1]
```diff
@@ -173,81 +173,14 @@ public long getTimeUntilNextRun()
...
-                // overusing and the target bitrate is too high compared to what
-                // we are receiving.
-                updateEstimate(nowMs);
-            }
-        }
-        } // synchronized (critSect)
+    public RawPacket reverseTransform(RawPacket pkt){
+        /**
+         * @TODO Findout what to do with the interArrivalTime in this method
+         */
```
Comment (bgrozev) [inline src/org/jitsi/impl/neomedia/rtp/remotebitrateestimator/RemoteBitrateEstimatorSingleStream.java]: I don't think we need the value in reverseTransform, so we can just ignore it.

File: src/org/jitsi/impl/neomedia/rtp/remotebitrateestimator/RemoteBitrateEstimatorSingleStream.java [hunk 2]
```diff
@@ -173,81 +173,14 @@ public long getTimeUntilNextRun()
...
+        long interArrivalTime = getInterArrivalDelayVariation (pkt.getSSRC(),
+        pkt.getTimestamp(),
+        System.currentTimeMillis(),
+        pkt.getLength());
```
Comment (bgrozev) [inline src/org/jitsi/impl/neomedia/rtp/remotebitrateestimator/RemoteBitrateEstimatorSingleStream.java]: This used to be getPayloadLength(), but I'm not sure what we want. I think if you're going to use the value as an encoder's target bitrate (as in webrtc.org) then you want the payload length only. But our bitrate calculations include RTP headers.
cc @gpolitis

File: src/org/jitsi/impl/neomedia/rtp/remotebitrateestimator/RemoteBitrateEstimatorSingleStream.java [hunk 3]
```diff
@@ -399,4 +332,127 @@ public Detector(
             this.detector = new OveruseDetector(options);
         }
     }
+
+
+    /**
+     * {@Link rtcpTCCBitrateEstimator} computes the current tcc bitrate for an
```
Comment (bgrozev) [inline src/org/jitsi/impl/neomedia/rtp/remotebitrateestimator/RemoteBitrateEstimatorSingleStream.java]: s/Link/link/

File: src/org/jitsi/impl/neomedia/rtp/remotebitrateestimator/RemoteBitrateEstimatorSingleStream.java [hunk 4]
```diff
@@ -399,4 +332,127 @@ public Detector(
...
+     * rtp packet.
+     * @param ssrc_ is the related source identifier of the sender
+     *             (for instance video-bridge)
+     * @param sendTime is the time the rtp packet was sent
+     * @param receivedTime is the arrival time on the reportedTCC Packet
+     * @param packetLength is the length of the rtp packet
+     * @return a bitrate value
```
Comment (bgrozev) [inline src/org/jitsi/impl/neomedia/rtp/remotebitrateestimator/RemoteBitrateEstimatorSingleStream.java]: Can we be more specific as to what we are returning?

File: src/org/jitsi/impl/neomedia/rtp/remotebitrateestimator/RemoteBitrateEstimatorSingleStream.java [hunk 5]
```diff
@@ -399,4 +332,127 @@ public Detector(
...
     }
+
+
+    /**
+     * {@Link rtcpTCCBitrateEstimator} computes the current tcc bitrate for an
+     * rtp packet.
+     * @param ssrc_ is the related source identifier of the sender
+     *             (for instance video-bridge)
+     * @param sendTime is the time the rtp packet was sent
+     * @param receivedTime is the arrival time on the reportedTCC Packet
```
Comment (bgrozev) [inline src/org/jitsi/impl/neomedia/rtp/remotebitrateestimator/RemoteBitrateEstimatorSingleStream.java]: Can you please specify what the format/clock for these two is?

File: src/org/jitsi/impl/neomedia/rtp/remotebitrateestimator/RemoteBitrateEstimatorSingleStream.java [hunk 6]
```diff
@@ -399,4 +332,127 @@ public Detector(
...
+    }
+
+
+    /**
+     * {@Link rtcpTCCRemoteBitrateEstimator} estimates the remote bitrate
+     * of the receiver from received rtcp tcc feedback packet parameters.
+     * This acts for the send-side of the bridge
+     * @param ssrc_ is the SSRC of the source
+     * @param sendTime is the time the packet was sent by the bridge
+     * @param receivedTime is the time recorded by the receiver
```
Comment (bgrozev) [inline src/org/jitsi/impl/neomedia/rtp/remotebitrateestimator/RemoteBitrateEstimatorSingleStream.java]: Same as above, make it clear that one is an RTP timestamp, the other is a local timestamp

**Comment by bgrozev** [comment]
@jcchuks Lets discuss this one when we next talk (so don't spend time switching back to it to fix my comments just yet). I think the approach I suggested might not be great, and we could just start using the abs-send-time RBE when you've got it.

**Comment by jcchuks** [comment]
blockquote, div.yahoo_quoted { margin-left: 0 !important; border-left:1px #715FFA solid !important; padding-left:1ex !important; background-color:white !important; } Alright 


Sent from Yahoo Mail for iPhone


On Thursday, June 22, 2017, 12:05 PM, bgrozev <notifications@github.com> wrote:


@jcchuks Lets discuss this one when we next talk (so don't spend time switching back to it to fix my comments just yet). I think the approach I suggested might not be great, and we could just start using the abs-send-time RBE when you've got it.

**Comment by jcchuks** [comment]
We are yet to discuss this. I'll take note of this.

**Pull request closed without merging**